src/Sylius/Bundle/CoreBundle/EventListener/SitemapListener.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\EventListener;
+
+use Symfony\Component\Routing\RouterInterface;
+use Presta\SitemapBundle\Service\SitemapListenerInterface;
+use Presta\SitemapBundle\Event\SitemapPopulateEvent;
+use Presta\SitemapBundle\Sitemap\Url\UrlConcrete;
+use \Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * Sitemap listener.
+ *
+ * @author Bartosz Siejka <bartosz.siejka@lakion.com>
+ */
+class SitemapListener implements SitemapListenerInterface
+{
+    private $event;
+    private $router;
+    private $dynamicRouter;
+    private $productRepository;
+    private $taxonRepository;
+
+    public function __construct(RouterInterface $router, RouterInterface $dynamicRouter, RepositoryInterface $productRepository, RepositoryInterface $taxonRepository)
+    {
+        $this->router = $router;
+        $this->dynamicRouter = $dynamicRouter;
+        $this->productRepository = $productRepository;
+        $this->taxonRepository = $taxonRepository;
+    }
+
+    public function populateSitemap(SitemapPopulateEvent $event)
+    {
+        $this->event = $event;
+        $section = $event->getSection();
+
+        if (is_null($section) || $section == 'default') {
+            $homepage = $this->router->generate('sylius_homepage', array(), true);
+            $products = $this->productRepository->findAll();
```

Comment:
To not kill big apps, this should be called only before loop and destroyed just after.